contracts/core-token-vesting/src/contract.rs

```diff
@@ -80,9 +84,178 @@
+) -> Result<Response, ContractError> {
+    let mut res = vec![];
+
+    let mut campaign = CAMPAIGN
+        .load(deps.storage)
+        .map_err(|_| StdError::generic_err("Failed to load campaign data"))?;
+
+    if campaign.owner != info.sender
+        && !campaign.managers.contains(&info.sender.into())
+    {
+        return Err(StdError::generic_err("Unauthorized").into());
+    }
+
+    if !campaign.is_active {
+        return Err(StdError::generic_err("Campaign is not active").into());
+    }
+
+    let total_requested: Uint128 = requests.iter().map(|req| req.amount).sum();
+    if total_requested > campaign.unallocated_amount {
+        return Err(
+            StdError::generic_err("Insufficient funds for all rewards").into()
+        );
+    }
+
+    let master_address = Some(campaign.owner.clone());
+    let mut attrs: Vec<Attribute> = vec![];
+
+    for req in requests {
+        // update the vesting amount inside the vesting schedule
+        let mut vesting_schedule = campaign.vesting_schedule.clone();
+
+        if let VestingSchedule::LinearVesting {
+            ref mut vesting_amount,
+            ..
+        } = vesting_schedule
+        {
+            *vesting_amount = req.amount;
+        }
+
+        let result = register_vesting_account(
+            deps.storage,
+            env.block.time,
+            master_address.clone(),
+            req.user_address.clone(),
+            campaign.denom.clone(),
+            req.amount,
+            vesting_schedule,
+        );
+
+        if let Ok(response) = result {
+            attrs.extend(response.attributes);
+            res.push(RewardUserResponse {
+                user_address: req.user_address.clone(),
+                success: true,
+                error_msg: "".to_string(),
+            });
+        } else {
+            let error = result.err().unwrap();
+            res.push(RewardUserResponse {
+                user_address: req.user_address.clone(),
+                success: false,
+                error_msg: "Failed to register vesting account: ".to_string()
+                    + &error.to_string(),
+            });
+        }
+    }
+
+    campaign.unallocated_amount = campaign.unallocated_amount - total_requested;
+    CAMPAIGN.save(deps.storage, &campaign)?;
+
+    Ok(Response::new()
+        .add_attributes(attrs)
+        .add_attribute("method", "reward_users")
+        .set_data(to_json_binary(&res).unwrap()))
+}
+
+fn create_campaign(
+    deps: DepsMut,
+    info: MessageInfo,
+    vesting_schedule: VestingSchedule,
+    campaign_name: String,
+    campaign_description: String,
+    managers: Vec<String>,
+) -> Result<Response, ContractError> {
+    if CAMPAIGN.may_load(deps.storage)?.is_some() {
+        return Err(StdError::generic_err("Campaign already exists").into());
+    }
+
+    if info.funds.len() != 1 {
+        return Err(StdError::generic_err("one denom sent required").into());
+    }
+
+    let coin = info.funds.get(0).unwrap();
```

Comment:
I think doing 

    let coin = info.funds.get(0)?;

is a little bit more controlled. If something fails in the other it panics and I wonder what the error is in those cases.